### PR TITLE
Refactor Hyperoblic,hyperbolic coords

### PR DESCRIPTION
**Pull request by SaitejaUtpala**

### CODE REVIEW

**Comment by SaitejaUtpala** [comment]
I used some of the code in #1348 , (cause PR is not merged yet). I moved methods from hyperoblic coords to hyperboloid

File: tests/tests_geomstats/test_hyperbolic.py [hunk 1]
```diff
@@ -18,478 +19,402 @@
...
-        result = log
-
-        self.assertAllClose(gs.shape(result), (n_samples, dim + 1))
-        self.assertAllClose(result, expected)
-
-        result = h5_metric.exp(tangent_vec=log, base_point=base_point)
-        expected = point
+            return self.generate_tests(smoke_data)
+
+        def extrinsic_to_intrinsic_coords_rasises_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:68]: TYPO

File: tests/tests_geomstats/test_hyperbolic.py [hunk 2]
```diff
@@ -18,478 +19,402 @@
...
+                self.n_vecs_list,
+                is_tangent_atol,
+            )
+
+        def projection_belongs_data(self):
+            return self._projection_belongs_data(
+                self.space_args_list,
+                self.shape_list,
+                self.n_samples_list,
+                belongs_atol=gs.atol * 1000,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:107]: Why 1000? it was 100 in the former tests

File: tests/tests_geomstats/test_hyperbolic.py [hunk 3]
```diff
@@ -518,46 +438,63 @@ def test_scaled_squared_norm(self):
...
+        dst_ball = ball_manifold.metric.dist(x_ball, y_ball)
+        dst_extr = space.metric.dist(x_extr, y_extr)
+        self.assertAllClose(dst_ball, dst_extr)
 
-        results = helper.test_parallel_transport(space, metric, shape)
-        for res in results:
-            self.assertTrue(res)
+    def test_distance_ball_extrinsic_intrinsic(self, dim, x_intrinsic, y_intrinsic):
 
-    def test_projection_and_belongs(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:552]: ok

File: tests/tests_geomstats/test_hyperbolic.py [hunk 4]
```diff
@@ -18,478 +19,402 @@
...
+        def exp_belongs_data(self):
+            return self._exp_belongs_data(
+                self.metric_args_list,
+                self.space_list,
+                self.shape_list,
+                self.n_samples_list,
+                belongs_atol=gs.atol * 1000,
+            )
+
+        def log_is_tangent_data(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:280]: there was a smoke case for this one

File: tests/tests_geomstats/test_hyperbolic.py [hunk 5]
```diff
@@ -518,46 +438,63 @@ def test_scaled_squared_norm(self):
...
+    def test_scaled_dist(self, dim, scale, point_a, point_b):
+        default_space = Hyperboloid(dim=dim)
+        scaled_space = Hyperboloid(dim=dim, scale=scale)
+        distance_default_metric = default_space.metric.dist(point_a, point_b)
         distance_scaled_metric = scaled_space.metric.dist(point_a, point_b)
         result = distance_scaled_metric
         expected = scale * distance_default_metric
         self.assertAllClose(result, expected)
 
-    def test_is_tangent(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:534]: missing (for log is tangent)

File: tests/tests_geomstats/test_hyperbolic.py [hunk 6]
```diff
@@ -18,478 +19,402 @@
...
-        vector = gs.array([0.001, 0.0, -0.00001, -0.00003])
-        tangent_vec = self.space.to_tangent(vector=vector, base_point=base_point)
-        exp = self.metric.exp(tangent_vec=tangent_vec, base_point=base_point)
-
-        result = self.metric.dist(base_point, exp)
-        sq_norm = self.metric.embedding_metric.squared_norm(tangent_vec)
-        expected = sq_norm
-        self.assertAllClose(result, expected, atol=1e-2)
-
-    def test_geodesic_and_belongs(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:431]: keep this one as a smoke test (small initial vector: that should belongs with a good atol)

File: tests/tests_geomstats/test_hyperbolic.py [hunk 7]
```diff
@@ -18,478 +19,402 @@
...
-        geodesic = self.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec
-        )
-
-        t = gs.linspace(start=0.0, stop=1.0, num=n_geodesic_points)
-        points = geodesic(t)
-        result = gs.all(self.space.belongs(points))
-        self.assertTrue(result)
-
-    def test_geodesic_and_belongs_large_initial_velocity(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:448]: also keep this one (larger initial vector: that belongs with a bigger atol)

File: tests/tests_geomstats/test_hyperbolic.py [hunk 8]
```diff
@@ -18,478 +19,402 @@
...
-        base_point = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        vector = gs.array([2.0, 1.0, 1.0, 1.0])
-        vector = self.space.to_tangent(vector=vector, base_point=base_point)
-        exp = self.metric.exp(tangent_vec=vector, base_point=base_point)
-        result = self.metric.log(point=exp, base_point=base_point)
-
-        expected = vector
-        self.assertAllClose(result, expected)
-
-    def test_dist(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:412]: missing?

Comment (ninamiolane) [reply tests/tests_geomstats/test_hyperbolic.py:412]: could go in RiemannianMetricParametrizer

File: tests/tests_geomstats/test_hyperbolic.py [hunk 9]
```diff
@@ -18,478 +19,402 @@
...
+        metric = self.metric(dim)
+        minkowki_space = Minkowski(dim + 1)
+        result = metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
+        expected = minkowki_space.metric.inner_product(
             tangent_vec_a, tangent_vec_b, base_point
         )
-
-        self.assertAllClose(result, expected)
-
-    def test_squared_norm_and_squared_dist(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:347]: missing?

File: tests/tests_geomstats/test_hyperbolic.py [hunk 10]
```diff
@@ -18,478 +19,402 @@
...
-        """
-        point_a = gs.array([2.0, 1.0, 1.0, 1.0])
-        point_b = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        log = self.metric.log(point=point_a, base_point=point_b)
-        result = self.metric.squared_norm(vector=log)
-        expected = self.metric.squared_dist(point_a, point_b)
-
-        self.assertAllClose(result, expected)
-
-    def test_norm_and_dist(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:360]: ok to skip this test

File: tests/tests_geomstats/test_hyperbolic.py [hunk 11]
```diff
@@ -18,478 +19,402 @@
...
-        """
-        point_a = gs.array([2.0, 1.0, 1.0, 1.0])
-        point_b = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        log = self.metric.log(point=point_a, base_point=point_b)
-        result = self.metric.norm(vector=log)
-        expected = self.metric.dist(point_a, point_b)
-
-        self.assertAllClose(result, expected)
-
-    def test_log_and_exp_edge_case(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:373]: missing?

File: tests/tests_geomstats/test_hyperbolic.py [hunk 12]
```diff
@@ -18,478 +19,402 @@
...
-        point_intrinsic = base_point_intrinsic + 1e-12 * gs.array([-1.0, -2.0, 1.0])
-        point = self.space.from_coordinates(point_intrinsic, "intrinsic")
-
-        log = self.metric.log(point=point, base_point=base_point)
-        result = self.metric.exp(tangent_vec=log, base_point=base_point)
-        expected = point
-
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_log_and_projection_to_tangent_space_general_case(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:394]: missing?

File: tests/tests_geomstats/test_hyperbolic.py [hunk 13]
```diff
@@ -18,478 +19,402 @@
...
-
-    def test_dist(self):
-        # Distance between a point and itself is 0.
-        point_a = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        point_b = point_a
-        result = self.metric.dist(point_a, point_b)
-        expected = 0
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_dist_and_projection_to_tangent_space(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_hyperbolic.py:420]: missing?

**Review by ninamiolane — COMMENTED** [review]
First batch of comments!

File: tests/tests_geomstats/test_hyperbolic.py (hunk 7 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_hyperbolic.py:448]: needs data generation that is struck in hypersphere...

File: tests/tests_geomstats/test_hyperbolic.py (hunk 12 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_hyperbolic.py:394]: nope already of part of exp-log composition

File: tests/tests_geomstats/test_hyperbolic.py (hunk 13 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_hyperbolic.py:420]: nope. already part of exp-log

File: tests/tests_geomstats/test_hyperbolic.py [hunk 14]
```diff
@@ -18,478 +19,402 @@
...
+                    point=gs.array([-1.0, 1.0, 0.0, 0.0]),
+                    expected=pytest.raises(ValueError),
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def random_point_belongs_data(self):
+            smoke_space_args_list = [(2,), (3,)]
+            smoke_n_points_list = [1, 2]
+            belongs_atol = gs.atol * 100000
```
[the remark this answers is not in the document]
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_hyperbolic.py:81]: but dimension is fixed before. that is not the case now

File: tests/tests_geomstats/test_hyperbolic.py (hunk 2 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_hyperbolic.py:107]: it was being tested for fixed dimension. now more dimensions

**Pull request merged**